dcpy/models/base.py

```diff
@@ -131,3 +132,49 @@ def from_yaml(cls, yaml_str: str, *, template_vars=None) -> typing.Self:
     def from_path(cls, path: Path, *, template_vars=None) -> typing.Self:
         with open(path, "r", encoding="utf-8") as raw:
             return cls.from_yaml(raw.read(), template_vars=template_vars)
+
+
+class ModelWithDataFrame(BaseModel, arbitrary_types_allowed=True):
+    def __eq__(self, other):
+        """
+        pandas DataFrames make simple equality checks a pain
+        naively checking equality of df-containing model produces 'ValueError: The truth value of a DataFrame is ambiguous'
+        therefore, df.equals() must be used
+
+        Only supported types so far are
+        - DataFrame
+        - dict[Any, DataFrame]
+
+        This could be extended to include other parameterized generic types
+        """
+
+        if type(other) is not type(self):
```

Comment:
speaking of footguns that Pydantic avoids, this feels like the most common one I encounter. (since type(myRegularClass)==`type`)